### PR TITLE
Implement changes to accomodate addition of data request byte

### PR DESCRIPTION
**Pull request by andrew103**
Also removes the recursive functionality of the serial reader and replaces it with returning -1 for all values addressing #7.

### CODE REVIEW

File: trajectory_executor/com_serial.py [hunk 1]
```diff
@@ -175,15 +174,30 @@ def sensor_data_request_callback(self, request: SensorDataRequest, response: Sen
...
+            response.yaw = -1
+            response.l_hip = -1
+            response.l_knee = -1
+            response.r_hip = -1
+            response.r_knee = -1
+            response.l_shoulder = -1
+            response.l_elbow = -1
+            response.r_shoulder = -1
+            response.r_elbow = -1
+            response.at_goal = -1
```
Comment (goobta) [inline trajectory_executor/com_serial.py]: What happens if the response position is at -1?

File: trajectory_executor/com_serial.py [hunk 2]
```diff
@@ -175,15 +174,30 @@ def sensor_data_request_callback(self, request: SensorDataRequest, response: Sen
...
+            break
 
     return response
 
   def send_joint_angles_callback(self, msg: JointAngles):
     """Send new joint angles to Arduino for execution"""
     # Split joint angles into two numbers since max value of a byte is 255
     ja_bytes = []
+    ja_bytes.append(0) # data request byte
```
Comment (goobta) [inline trajectory_executor/com_serial.py:200]: Not 100% sure I'm reading this correctly, but this byte is basically just a flag telling if it's a response or a request, correct?

**Review by goobta — COMMENTED** [review]
Just a few comments.

Also do we have tests for this? Especially since this is a state machine in pure python/ROS, we can def unit test the living fuck outta this.

I feel like we should resolve the issues on at least one side fully. If we only go off of functional testing between ROS and the arduino--that's two repos on two different branches with three people working on it spread across the country. If we get a small, but breaking change like the checksum is off slightly or an error is being supressed--we'll never find out that bug. 

Since we have this great state machine, we could quite literally write a test for every single state. Hopefully `parameterized` should be able to basically make writing those tests a breeze. The main benefit of having a fully tested ROS node is that we will then have  'ground-truth' that when we running the arduino against it, we know all of the errors are on the arduino side and not the ROS one.

@mahajanrevant thoughts? Gonna let you approve cause once again I've never actually used ros.

File: trajectory_executor/com_serial.py (hunk 1 above)
Comment (andrew103) [reply trajectory_executor/com_serial.py]: I'm thinking that the executor node will check is if any of the values are -1 and if they are then it knows there was an error and sends another request or something.

File: trajectory_executor/com_serial.py (hunk 2 above)
Comment (andrew103) [reply trajectory_executor/com_serial.py:200]: Correct. 0: sending joint angles. 1+: requesting sensor data

**Comment by andrew103** [comment]
@agupta231 At the moment, we just have the basic testing node `serial_test.py`. As far as I can tell using that testing node, everything is working as expected. With that said, yes having more in-depth tests would be a good idea and ROS has a structure in place to create tests. However, and I believe this was mentioned in a previous PR, doing testing on the `com_serial` node could prove to be a complex undertaking considering that the node deals with outgoing and incoming serial data. You couldn't unit test the joint angles output without having some sort of serial port snooper and you can't unit test the sensor data request without having a snooper and trusting that the Arduino runs properly and provides the correct response.

With that said, we should make an issue and trello card for unit testing this node to later come back to. For now, this PR and the arduino PR are meant as hotfix patches for an existing issue rather than a feature update.

File: trajectory_executor/com_serial.py (hunk 1 above)
Comment (goobta) [reply trajectory_executor/com_serial.py]: Ahh ok so they're just valued in `[0, 360]` degrees, makes sense!

File: trajectory_executor/com_serial.py (hunk 2 above)
Comment (mahajanrevant) [reply trajectory_executor/com_serial.py:200]: Let's declare a variable that holds this value? Lets also add a comment to explain what is going on

**Review by mahajanrevant — COMMENTED** [review]
One small comment

**Comment by andrew103** [comment]
Yo @mahajanrevant @agupta231 I just realized this PR hasn't been merged in yet. Can this get approved real quick so that it can be merged into master? It's critical to the master branch working.

**Review by mahajanrevant — APPROVED** [review]
Approving with a minor issue created

**Pull request merged**